app/navigators/ChargeENavigator.scala

```diff
@@ -116,14 +99,67 @@ class ChargeENavigator @Inject()(val dataCacheConnector: UserAnswersCacheConnect
     case TaxQuarterReportedAndPaidPage(ChargeTypeAnnualAllowance, index, schemeIndex) =>
       controllers.mccloud.routes.ChargeAmountReportedController
         .onPageLoad(ChargeTypeAnnualAllowance, NormalMode, srn, startDate, accessType, version, index, schemeIndex)
-    case ChargeAmountReportedPage(ChargeTypeAnnualAllowance, index, _) =>
-      CheckYourAnswersController.onPageLoad(srn, startDate, accessType, version, index)
+    case ChargeAmountReportedPage(ChargeTypeAnnualAllowance, index, schemeIndex) =>
+      routeFromChargeAmountReportedPage(ua, srn, startDate, accessType, version, index, schemeIndex)
 
+    case AddAnotherPensionSchemePage(ChargeTypeAnnualAllowance, index, schemeIndex) =>
+      ua.get(AddAnotherPensionSchemePage(ChargeTypeAnnualAllowance, index, schemeIndex)) match {
```

Comment:
Move any blocks of code like this out of main routeMap method into separate method. This will just help keep this main method short and readable.

Reply:
Fixed.